CleverTapSDKTests/EventDatabase/CTEventDatabaseTests.m

```diff
@@ -0,0 +1,286 @@
+    [self.eventDatabase insertEvent:kEventName
+                normalizedEventName:self.normalizedEventName
+                           deviceID:kDeviceID];
+    
+    NSString *normalizedEventName = [CTUtils getNormalizedName:@"TesT   EveNT 1"];
+    BOOL eventExists = [self.eventDatabase eventExists:normalizedEventName forDeviceID:kDeviceID];
+    XCTAssertFalse(eventExists);
+    
+    normalizedEventName = [CTUtils getNormalizedName:@"Test.Event"];
+    eventExists = [self.eventDatabase eventExists:normalizedEventName forDeviceID:kDeviceID];
+    XCTAssertFalse(eventExists);
+}
+
+- (void)testUpdateEventSuccess {
+    [self.eventDatabase insertEvent:kEventName
+                normalizedEventName:self.normalizedEventName
+                           deviceID:kDeviceID];
+    
+    BOOL updateSuccess = [self.eventDatabase updateEvent:self.normalizedEventName forDeviceID:kDeviceID];
+    XCTAssertTrue(updateSuccess);
+}
+
+- (void)testUpsertEventSuccessWhenInsert {
+    BOOL upsertSuccess = [self.eventDatabase upsertEvent:kEventName
+                                     normalizedEventName:self.normalizedEventName
+                                                deviceID:kDeviceID];
+    XCTAssertTrue(upsertSuccess);
+}
+
+- (void)testUpsertEventSuccessWhenUpdate {
+    [self.eventDatabase insertEvent:kEventName
+                normalizedEventName:self.normalizedEventName
+                           deviceID:kDeviceID];
+
+    BOOL upsertSuccess = [self.eventDatabase upsertEvent:kEventName
+                                     normalizedEventName:self.normalizedEventName
+                                                deviceID:kDeviceID];
+    XCTAssertTrue(upsertSuccess);
+    
+    NSInteger eventCount = [self.eventDatabase getEventCount:self.normalizedEventName deviceID:kDeviceID];
+    XCTAssertEqual(eventCount, 2);
+}
+
+- (void)testGetCountForEventName {
+    [self.eventDatabase insertEvent:kEventName
+                normalizedEventName:self.normalizedEventName
+                           deviceID:kDeviceID];
+    
+    [self.eventDatabase updateEvent:self.normalizedEventName forDeviceID:kDeviceID];
+    
+    // count should be 2.
+    NSInteger eventCount = [self.eventDatabase getEventCount:self.normalizedEventName deviceID:kDeviceID];
+    XCTAssertEqual(eventCount, 2);
+}
+
+- (void)testGetCountForEventNameNotExists {
+    // count should be 0.
+    NSInteger eventCount = [self.eventDatabase getEventCount:self.normalizedEventName deviceID:kDeviceID];
+    XCTAssertEqual(eventCount, 0);
+}
+
+- (void)testFirstTimestampForEventName {
+    NSInteger currentTs = (NSInteger)[[NSDate date] timeIntervalSince1970];
+    [self.eventDatabase insertEvent:kEventName
+                normalizedEventName:self.normalizedEventName
+                           deviceID:kDeviceID];
+    
+    NSInteger firstTs = [self.eventDatabase getFirstTimestamp:self.normalizedEventName deviceID:kDeviceID];
+    XCTAssertEqual(firstTs, currentTs);
```

Comment:
This can easily fail if the `insertEvent` takes more than a second or there is code executed between setting `currentTs` and calling insert.

This is valid for other tests as well.

Reply:
Updated tests.